datasets/hr/public_officials/crawler.py

```diff
@@ -0,0 +1,126 @@
+        legal_entity = make_legal_entity(context, legal_entity_name)
+        position_name = position_name or "Unspecified position"
+        position = h.make_position(
+            context,
+            position_name,
+            topics=None,
+            organization=legal_entity,
+        )
+
+        categorisation = categorise(context, position, is_pep=True)
+        occupancy = h.make_occupancy(
+            context,
+            person,
+            position,
+            no_end_implies_current=True,
+            categorisation=categorisation,
+            propagate_country=True,
+            start_date=convert_date(start_date),
+            end_date=convert_date(end_date),
+        )
+        if occupancy:
+            context.emit(position)
+            context.emit(occupancy)
+    context.audit_data(data)
+
+
+def make_person(context: Context, first_name: str, last_name: str) -> Entity:
+    person = context.make("Person")
+    person.id = context.make_slug(first_name, last_name)
+    person.add("firstName", first_name)
+    person.add("lastName", last_name)
+    person.add("country", "LT")
+    person.add("topics", "role.pep")
+    return person
+
+
+def validate_headers(context: Context, headers: list) -> None:
+    """Raises AssertionError if columns are missing or not in the expected order.
+    Logs extra columns. Note: This CSV has duplicate column names.
+    """
+    error_message = f"Unexpected column headers: {headers}"
+    assert headers[: len(FIELD_NAMES)] == FIELD_NAMES, error_message
+    if len(headers) > len(FIELD_NAMES):
+        context.log.warn(f"Unexpected column headers: {headers[:len(FIELD_NAMES):]}")
+
+
+def extract_dict_keys_by_prefix(
+    data: dict, key_names: list[str], prefix: str
+) -> dict[str:Any]:
+    """Pops dict keys in `key_names` if the key starts with `prefix`.
+    Returns a new dict and removes keys from old dict"""
+    return {
+        k.removeprefix(prefix): data.pop(k) for k in key_names if k.startswith(prefix)
+    }
+
+
+def crawl(context: Context):
+    """Fetches the current CSV file and crawls each row, making persons, legal entities, and positions"""
+    file_path = context.fetch_resource("daily_csv_release", context.data_url)
+    context.export_resource(file_path, CSV, title=context.SOURCE_TITLE)
+    with open(file_path, encoding="utf-8-sig") as fh:
+        reader = csv.DictReader(fh, fieldnames=FIELD_NAMES, delimiter=";")
+        for row in reader:
+            person = make_person(context, row.pop("Ime"), row.pop("Prezime"))
+
+            affiliation_1 = extract_dict_keys_by_prefix(row, FIELD_NAMES, "Primarna ")
+            handle_affiliation_data(context, person, affiliation_1)
+
+            affiliation_2 = extract_dict_keys_by_prefix(row, FIELD_NAMES, "Sekundarna ")
+            handle_affiliation_data(context, person, affiliation_2)
+
+            context.audit_data(row)
+            context.emit(person, target=True)
```

Comment:
We only want to emit people in PEP datasets that are actually PEPs, and drop them when they leave our "PEP period" - the period for which we consider someone a PEP after they've left the position.

A pattern we sometimes use is to make a list of position and occupancy entities if any occupancies are created for the person, and only in that case, emit the person (and the occupancies and positions)

That way we also only emit positions for which there are persons who hold that position.

Even if we start off defaulting to assume all these positions are PEP positions (categorise(is_pep=True)) which seems reasonable looking at these positions, we might later categorise some as non-PEP, or some local gov person might not enter any new positions for 5 years but still be in the source file.